### PR TITLE
[FIX] *: cleanup manifest files

### PR DESCRIPTION
**Pull request by Julien00859**
*: base_address_city, base_address_extended, bus, crm, im_livechat,
   l10n_ae_pos, lunch, pos_restaurant_adyen, test_assetsbundle,
   test_converter, test_lint.

It is spelled `auto_install`, the `complexity` key is long gone, `qweb`
has been moved to `assets: {'web.assets_qweb': []}`. `js` and `css` are
long gone too, `maintainer` is redundant with `author` which is
"Odoo S.A." by default already, the `certificate` key is long gone.

### CODE REVIEW

File: addons/l10n_ae_pos/__manifest__.py
Comment (VincentSchippefilt) [inline addons/l10n_ae_pos/__manifest__.py:16]: was it working before ? because if this file wasn't integrated to the templates, there is a test missing !

Comment (william-andre) [reply addons/l10n_ae_pos/__manifest__.py:16]: People didn't bother with tests for localisation before it was integrated in the CI (since 15.0)

File: addons/pos_restaurant_adyen/__manifest__.py
Comment (xmo-odoo) [inline addons/pos_restaurant_adyen/__manifest__.py:15]: ping @pimodoo @odoo/rd-payment might be interested in pos_restaurant_adyen having never auto-installed.

**Review by xmo-odoo — APPROVED** [review]
:+1:

File: addons/l10n_ae_pos/__manifest__.py
Comment (pimodoo) [reply addons/l10n_ae_pos/__manifest__.py:16]: put it in 'point_of_sale.assets'

**Comment by sdegueldre** [comment]
Wouldn't it be a good idea to introduce errors/warnings when the manifest contains unknown keys? Seems like all of the things that this PR fixes could've been fixed immediately if that was the case.

**Comment by Julien00859** [comment]
> Wouldn't it be a good idea to introduce errors/warnings when the manifest contains unknown keys? Seems like all of the things that this PR fixes could've been fixed immediately if that was the case.

The problem is that there are *some* manifest files where there we use "undefined keys" to store some extra info. e.g. while some community contributors were clever and added comments on top of the manifest for the legal stuff:

https://github.com/odoo/enterprise/blob/669312e7d4ca5ecf577ea7835bb9698a24612f80/l10n_co_reports/__manifest__.py#L4-L9

Others store it in an undefined "maintainer" key:

https://github.com/odoo/odoo/blob/ad6168963dea3eb1f4712e3ce94ee58043af90db/addons/l10n_cn/__manifest__.py#L15

https://github.com/odoo/odoo/blob/ad6168963dea3eb1f4712e3ce94ee58043af90db/addons/l10n_ec/__manifest__.py#L37

This "maintainer" key is one example, there are a few others keys (10-ish IIRC-from-yesterday) often in modules maintained by the community. I am not too comfortable to remove/replace/comment those keys which would be a necessary step before asserting no unknown key is used.

The other important reason I'm against sending warnings for unknown keys is that there are keys that are module-specific. E.g the few keys related to website and website themes. We should allow modules to add their own keys to a global "valid keys" list, while it could be as simple as `VALID_KEYS = set(...)` `import odoo.modules.module; odoo.modules.module.add(...)` I'm not too sure we always import all modules source code *before* loading their manifest.

**Comment by Julien00859** [comment]
@robodoo r+

**Comment by william-andre** [comment]
> The other important reason I'm against sending warnings for unknown keys is that there are keys that are module-specific. E.g the few keys related to website and website themes. We should allow modules to add their own keys to a global "valid keys" list, while it could be as simple as `VALID_KEYS = set(...)` `import odoo.modules.module; odoo.modules.module.add(...)` I'm not too sure we always import all modules source code _before_ loading their manifest.

@Julien00859 You could add a key stating explicitly that uncommon keys are valid. Sort of like a `# pylint: disable=`.

**Comment by william-andre** [comment]
> `@robodoo r+`

Maybe you should let ppl react to the comment you just write before self reviewing yourself?

**Comment by sdegueldre** [comment]
You could also have a "metadata" key where people can store arbitrary things that are module specific, this key would be documented to have no special meaning to the odoo framework itself but can be used by code within the module.

**Comment by william-andre** [comment]
@antonylesuisse Can we remove self r+ right once and for all? It made sense when Odoo was smaller, but now you can always find someone to approve your PRs if needed.
90% of the time I see someone self reviewing, there are still pending comments or it has not been reviewed at all.

**Comment by xmo-odoo** [comment]
> Maybe you should let ppl react to the comment you just wrote before self reviewing yourself?

None of the comments has any relevance to the actual PR though, they're all about additional features which can be added later.

**Comment by william-andre** [comment]
> None of the comments has any relevance to the actual PR though

If what is the point of cleaning it if it is not also to prevent having it. Do these keys make sense or not? That is the question.

> they're all about additional features which can be added later

This is a master task, what is the issue? ~~Is~~ Was there a rush to merge this PR?

**Comment by Julien00859** [comment]
The "metadata" key (that I would name "extra" as some keys not really meta but actually in use in some specif modules) is a good idea. Maybe we could test for some keys we know are errors (qweb, autoinstall, auto-install, maybe levenstein distance ??) and emit a warning if they are detected "you have 'autoinstall' in your manifest, did you mean 'auto_install'?".

Is it worth it though? IMO testing for "sure to be broken" keys should be enough.

**Comment by sdegueldre** [comment]
> e.g. as far as I'm concerned this could well be a instead

@xmo-odoo I think you're missing a word here.

But yes, I did not mean this comment to be blocking for the PR, I think it's relevant to mention this here but this PR is just a net positive and there was no problem with merging it as is IMO.

@Julien00859 I think levenstein distance is overkill and there will inevitably be false positives, a whitelist is a simpler and more robust approach and I think the problem of "custom keys" is easily worked around in a multitude of ways without having to allow arbitrary keys in the manifest

**Comment by Xavier-Do** [comment]
This time you nicely fixed all manifest and people began complaining.

**Pull request closed without merging**
